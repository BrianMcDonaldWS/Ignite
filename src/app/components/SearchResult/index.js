/* eslint-disable react/no-danger */

import path from 'path';
import React from 'react';
import PropTypes from 'prop-types';
import makeClass from 'classnames';
import markdown from 'zeedown';
import { Link } from 'react-router-dom';

import styles from './searchResult.css';

const SearchResult = ({ fileName, baseURL, results, setResults }) => (
  <div key={fileName} className="card">
    <div key={fileName} className="card-content">
      <Link
        className={makeClass('subtitle', 'has-text-link', styles.title)}
        to={path.join(baseURL, fileName.replace('.md', '.html'))}
        onClick={() => setTimeout(() => setResults({}), 10)}
      >
        {fileName}
      </Link>

      {[...results].map(result => (
        <div key={result}>
          <p
            dangerouslySetInnerHTML={{
              __html: markdown(result)
            }}
          />
          <br />
        </div>
      ))}
    </div>
  </div>
);

SearchResult.propTypes = {
  setResults: PropTypes.func,
  fileName: PropTypes.string.isRequired,
  baseURL: PropTypes.string,
  results: PropTypes.array
};

SearchResult.defaultProps = {
  setResults: () => {},
  baseURL: process.env.baseURL,
  results: []
};

export default SearchResult;
