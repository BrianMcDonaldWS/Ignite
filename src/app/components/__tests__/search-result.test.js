import React from 'react';

import SearchResult from '../SearchResult';
import renderToJson from './utils/render-to-json';

test('SearchResult', () => {
  expect(
    renderToJson(
      <SearchResult
        baseURL="/"
        fileName="foo.md"
        results={['text results for foo.md']}
      />
    )
  ).toMatchSnapshot();
});
