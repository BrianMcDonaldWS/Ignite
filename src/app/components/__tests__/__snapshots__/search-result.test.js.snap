// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <ForwardRef
      className="subtitle has-text-link title"
      onClick={[Function]}
      to="/foo.html"
    >
      foo.md
    </ForwardRef>
    <div>
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "text results for foo.md",
          }
        }
      />
      <br />
    </div>
  </div>
</div>
`;
